### PR TITLE
feat: draw new dashboard widget container

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci`, minor refactoring, etc.)
- [ ] Need discussion
- [x] Not that difficult
- [ ] Approved feature branch merge to master

### Description
스크린샷 2024-06-12 오후 9 31 55

### CODE REVIEW

**Review by piggggggggy — APPROVED** [review]
LGTM~

File: apps/web/src/services/dashboards/stores/dashboard-detail-info-store.ts [hunk 1]
```diff
@@ -280,11 +254,17 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
         setVariablesInitMap(_variablesInitMap);
 
         // widget info states
-        const _dashboardWidgetInfoList = flattenDeep(_dashboardInfo?.layouts.map((layout) => layout.widgets ?? [])).map((info) => ({
-            ...info,
-            widget_key: info.widget_key ?? getRandomId(),
-        })) ?? [];
-        setDashboardWidgetInfoList(_dashboardWidgetInfoList);
+        const _dashboardVersion = _dashboardInfo.version;
+        if (_dashboardVersion === '1.0') {
```
Comment (sulmoJ) [inline apps/web/src/services/dashboards/stores/dashboard-detail-info-store.ts:258]: annotation

File: apps/web/src/services/dashboards/components/DashboardControlButtons.vue [hunk 2]
```diff
@@ -1,48 +1,104 @@
...
-        <p-button icon-left="ic_dashboard-customize"
-                  style-type="tertiary"
-                  @click="handleClickCustomize"
+        <p-button icon-left="ic_plus"
+                  style-type="substitutive"
+                  :loading="state.loading"
+                  @click="handleAddWidget"
         >
-            {{ $t('DASHBOARDS.DETAIL.CUSTOMIZE') }}
+            Add Widget
```
Comment (sulmoJ) [inline apps/web/src/services/dashboards/components/DashboardControlButtons.vue:95]: translation

File: apps/web/src/services/dashboards/components/DashboardControlButtons.vue [hunk 3]
```diff
@@ -1,48 +1,104 @@
...
         </p-button>
         <p-button icon-left="ic_duplicate"
-                  style-type="tertiary"
-                  @click="handleVisibleCloneModal"
+                  style-type="secondary"
+                  @click="handleClickWidgetReorder"
         >
-            {{ $t('DASHBOARDS.DETAIL.CLONE') }}
+            Widget Reorder
```
Comment (sulmoJ) [inline apps/web/src/services/dashboards/components/DashboardControlButtons.vue:101]: translation

File: apps/web/src/services/dashboards/components/DashboardWidgetContainerV2.vue [hunk 4]
```diff
@@ -173,71 +175,63 @@ const handleOpenWidgetOverlay = (widget: RefinedWidgetInfo) => {
...
-    // widgetDeleteState.visibleModal = false;
-    // widgetDeleteState.targetWidget = null;
+const handleDeleteModalConfirm = async () => {
+    // 1. remove from dashboard layouts
+    await dashboardDetailStore.deleteDashboardWidget(widgetDeleteState.targetWidget?.widget_id);
+    // 2. delete widget
+    await deleteWidget(widgetDeleteState.targetWidget?.widget_id as string);
+    // 3. close modal
+    widgetDeleteState.visibleModal = false;
+    widgetDeleteState.targetWidget = null;
```
Comment (sulmoJ) [inline apps/web/src/services/dashboards/components/DashboardWidgetContainerV2.vue:229]: 👍

File: apps/web/src/services/dashboards/stores/dashboard-detail-info-store.ts [hunk 5]
```diff
@@ -98,7 +98,8 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
         templateId: 'blank', // "templateId" exists in new dashboard, but not in existing dashboard.
         templateType: 'MANAGED' as TemplateType,
         // widget info states
-        dashboardWidgetInfoList: [] as Array<DashboardLayoutWidgetInfo|DashboardLayoutWidgetInfoV2>,
+        dashboardWidgetInfoList: [] as DashboardLayoutWidgetInfo[], // only for 1.0 dashboard
+        dashboardLayouts: [] as DashboardLayout[], // only for 2.0 dashboard
```
Comment (sulmoJ) [inline apps/web/src/services/dashboards/stores/dashboard-detail-info-store.ts:102]: annotation

File: apps/web/src/services/dashboards/stores/dashboard-detail-info-store.ts [hunk 6]
```diff
@@ -149,6 +150,9 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
     const setDashboardWidgetInfoList = (dashboardWidgetInfoList: DashboardLayoutWidgetInfo[]) => {
         state.dashboardWidgetInfoList = dashboardWidgetInfoList;
     };
```
Comment (sulmoJ) [inline apps/web/src/services/dashboards/stores/dashboard-detail-info-store.ts:152]: annotation

**Review by sulmoJ — APPROVED** [review]
LGTM~ Questions were left in slack.

**Pull request merged**